connector/src/main/java/org.apache.flink/connector/nebula/table/NebulaDynamicTableFactory.java

```diff
@@ -80,11 +96,140 @@ public class NebulaDynamicTableFactory implements DynamicTableSourceFactory,
+        final FactoryUtil.TableFactoryHelper helper =
+                FactoryUtil.createTableFactoryHelper(this, context);
+        final ReadableConfig config = helper.getOptions();
+
+        final DataType producedDataType =
+                context.getCatalogTable().getResolvedSchema().toPhysicalRowDataType();
+
+        helper.validate();
+        validateConfigOptions(config);
+        return new NebulaDynamicTableSink(
+                getClientOptions(config), getExecutionOptions(context, config), producedDataType);
+    }
+
+    private void validateConfigOptions(ReadableConfig config) {
+        if (!config.getOptional(METAADDRESS).isPresent()) {
+            throw new IllegalArgumentException(
+                    String.format("The value of '%s' option should not be null",
+                            METAADDRESS.key()));
+        }
+
+        if (!config.getOptional(GRAPHADDRESS).isPresent()) {
+            throw new IllegalArgumentException(
+                    String.format("The value of '%s' option should not be null",
+                            GRAPHADDRESS.key()));
+        }
+
+        if (!config.getOptional(USERNAME).isPresent()) {
+            throw new IllegalArgumentException(
+                    String.format("The value of '%s' option should not be null",
+                            USERNAME.key()));
+        }
+
+        if (!config.getOptional(PASSWORD).isPresent()) {
+            throw new IllegalArgumentException(
+                    String.format("The value of '%s' option should not be null", PASSWORD.key()));
+        }
+
+        if (!config.getOptional(GRAPH_SPACE).isPresent()) {
+            throw new IllegalArgumentException(
+                    String.format("The value of '%s' option should not be null",
+                            GRAPH_SPACE.key()));
+        }
+
+        if (config.get(CONNECT_TIMEOUT) < 0) {
+            throw new IllegalArgumentException(
+                    String.format("The value of '%s' option should not be negative, but is %s.",
+                            CONNECT_TIMEOUT.key(), config.get(CONNECT_TIMEOUT)));
+        }
+
+        if (config.get(CONNECT_RETRY) < 0) {
+            throw new IllegalArgumentException(
+                    String.format("The value of '%s' option should not be negative, but is %s.",
+                            CONNECT_RETRY.key(), config.get(CONNECT_RETRY)));
+        }
+
+        if (config.get(TIMEOUT) < 0) {
+            throw new IllegalArgumentException(
+                    String.format("The value of '%s' option should not be negative, but is %s.",
+                            TIMEOUT.key(), config.get(TIMEOUT)));
+        }
+
+        if (config.get(EXECUTE_RETRY) < 0) {
+            throw new IllegalArgumentException(
+                    String.format("The value of '%s' option should not be negative, but is %s.",
+                            EXECUTE_RETRY.key(), config.get(EXECUTE_RETRY)));
+        }
```

Comment:
maybe we can keep just one parameter for `timeout` and remove the `retry` parameter, because java client 3.0 just exposes `timeout` for `NebulaPoolConfig`.

Reply:
These are the existing codes. I am fine with review and removed EXECUTE_RETRY related codes.